src/plain.jl

```diff
@@ -606,6 +606,27 @@ function h5open(f::Function, args...)
     end
 end
 
+if VERSION >= v"0.4"
+    function h5rewrite(f::Function, filename::AbstractString, args...)
+        tmppath,tmpio = mktemp(dirname(filename))
+        close(tmpio)
+
+        try
+            fid = h5open(tmppath, "w", args...)
```

Comment:
It would be better to use `H5Freopen` here with `tmpio`'s file descriptor as argument; this avoids a race condition.

I don't know whether `H5Freopen` is accessible from Julia, or whether this race condition is important in practice.